src/Makefile

```diff
@@ -282,3 +282,33 @@ clean ::
 	\rm -f ../analysis/swig_generated/*.so
 	\rm -f *.pyc
 	\rm -f ../analysis/swig_generated/SphSim.py
+	\rm -f $(PWD)/src/*.d
+
+
+# Handle the dependencies using black magic
+get_dir_str 	= $(patsubst %/,%,$(dir $1))
+WHICH_GPP = which g++
+ifeq ($(findstring g++,$(WHICH_GPP)),g++)
+BUILD_DEPS = @set -e; g++ -M -MG $< |\
+	 sed 's;\($*\)\.o[ :]*;$(call get_dir_str, $@)/\1.o $@ : ;g' > $@ 
+else
+BUILD_DEPS = 
+endif
+
+CPP_TGT := $(patsubst %.o,  %.cpp, $(OBJ)) 
+CPP_TGT := $(patsubst chealpix.cpp, , $(CPP_TGT)) 
```

Comment:
patsubstr is using the list of objects to generate a .cpp file name. The dependency resolution checks to see if these files 1) exist and 2) have been modified. Since chealpix is a .c file, Make obviously won't find chealpix.cpp as it doesn't exist. Therefore, I just use patsubstr to remove it from the list. 

It would be neater to create a list of .cpp files and .c files that I could use for dependency resolution, from which the .o file list is created. But this method works and required a much smaller change.